azext_edge/edge/providers/check/mq.py

```diff
@@ -746,10 +1199,354 @@ def _evaluate_broker_diagnostics_service(
                     target_name=target_brokers,
                     namespace=namespace,
                     display=Padding(
-                        f"[cyan]{port.get('name')}[/cyan] "
-                        f"port [bright_blue]{port.get('port')}[/bright_blue] "
-                        f"protocol [cyan]{port.get('protocol')}[/cyan]",
+                        f"{colorize_string(port.get('name'))} "
+                        f"port {colorize_string(port.get('port'))} "
+                        f"protocol {colorize_string(port.get('protocol'))}",
                         (0, 0, 0, 16),
                     ),
                 )
             check_manager.add_display(target_name=target_brokers, namespace=namespace, display=NewLine())
+
+
+def _get_valid_references(kind: Union[Enum, str], namespace: Optional[str] = None) -> Dict[str, Any]:
+    result = {}
+    custom_objects = MQ_ACTIVE_API.get_resources(kind=kind, namespace=namespace)
+    if custom_objects:
+        objects: List[dict] = custom_objects.get("items", [])
+        for object in objects:
+            o: dict = object
+            metadata: dict = o.get("metadata", {})
+            name = metadata.get("name")
+            if name:
+                result[name] = True
+
+    return result
+
+
+def _display_sub_check_results(
+    check_manager: CheckManager,
+    target_name: str,
+    namespace: str,
+    sub_check_results: List[CheckResult],
+    parent_padding: int,
+    detail_level: int = ResourceOutputDetailLevel.summary.value,
+) -> None:
+    errors_displays = []
+    for result in sub_check_results:
+        # summary level will only show header and non-success results
+        # detail level will show all results with evaluation status
+        # verbose level will show all results
+        if (
+            detail_level == ResourceOutputDetailLevel.summary.value
+            and result.eval_status == CheckTaskStatus.error.value
+        ):
+            text = result.display.renderable.replace("- ", "")
```

Comment:
This might remove hyphens from other parts of displays - if we only care about hyphens on the outside of strings, we could also use `.strip("- ")`